doc/news/_preparation_next_release.md

```diff
@@ -145,6 +145,19 @@ and be assured that you can safely set a path value to the key. A more detailed
 
 - Added some basic unit tests _(Bernhard Denner)_
 
+### Quickdump
```

Comment:
Does quickdump also work as transport format for pluginprocess? It could be very interesting as we have performance difficulties for large key sets, see #2209.

Btw. it would be great if we find a BSD-licensed alternative to the "portable" library. Why are ntohl (or htobe64 if needed) and similar not enough?

Reply:
after the recently discussed changes to mmapstorage (copy data to non-regular files), it should be compatible too, fyi

Reply:
> Does quickdump also work as transport format for pluginprocess?

(In theory) Yes, quickdump is now 100% equivalent to dump. It even supports shared metakeys (similar to `keyCopyMeta` in dump).

> after the recently discussed changes to mmapstorage (copy data to non-regular files), it should be compatible too, fyi

I think `quickdump` would be more suitable for IPC. If pipes are involved `mmapstorage` has to do a lot of copying and in the end it has to fix all pointers.

> Btw. it would be great if we find a BSD-licensed alternative to the "portable" library.

I will look into that. The current files are simply taken from https://github.com/dvidelabs/flatcc. 

> Why are ntohl (or htobe64 if needed) and similar not enough?

Firstly, as noted in the README, I decided to use little-endian (except for the magic number) to store the data, because of performance advantages.

https://github.com/ElektraInitiative/libelektra/blob/6847f6831adad154c6b03e86ef53ae1a51eb3888/src/plugins/quickdump/README.md#L28

The "library" comes from Flatbuffers (a performance optimized serialization format), it uses the most performant conversion it can find. `htobe64` et al. are non-standard. If they are available (and no compiler built-in can be used) they will be used. In practice of course using a naive implementation with `<<` and `&` would be good enough too, because almost all modern systems are little-endian anyway.

Reply:
I found this public domain endianness helper file: https://gist.github.com/michaeljclark/3b4fd912f6fa8bb598b3 
It covers less cases of optimization, but it should work for our purposes.

I added it as `kdbendian.h` so other plugins/libraries can use it too.

Reply:
> Yes, quickdump is now 100% equivalent to dump.

So it is basically binarydump?

> It covers less cases of optimization, but it should work for our purposes. I added it as kdbendian.h so other plugins/libraries can use it too.

Thank you!

Reply:
> So it is basically binarydump?

Basically yes. Although `dump` technically uses binary files too, because it stores the null terminators of strings.